CMS-Enterprise/mint-app
# SQLFluff upgrade to v3

# NOREF

## Description

- Updates SQLFluff to the latest version
- Simplify config to specify [core rules](https://docs.sqlfluff.com/en/stable/reference/rules.html#core-rules) as our default, with some rules-specific configs on top of that
- Re-run SQLFluff with new version & config (which is why this touches so many files!)

## How to test this change

The main goal with testing this PR is to ensure that the linter runs consistently and behaves in a way that we find useful. For that reason, feel free to poke around with editing SQL files and re-running the linter as you feel necessary!

1.  Run `pre-commit run --all sqlfluff-fix` and ensure no files are touched and the command completes successfully (`echo $?` should return `0`!)
2. Edit a SQL File in a way that you think should break the linter config (mixing case of keywords, bad indentation, etc.)
3. Re run `pre-commit run --all sqlfluff-fix` and ensure the linter either fails or updates the file as expected (if an auto-fixable rule was broken)

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
